### ISSUE TITLE
Handle Option - missing generics for enum `Option`

### ISSUE THREAD

**Issue by tivrfoa** [tivrfoa/new-derive#2]
```rust
use new_derive::New;

#[derive(Debug, New)]
struct S2 {
	opt1: Option<String>,
}

fn main() {
	let s2 = S2::new(Some("ola".into()));
	dbg!(s2);
}
```

Error:

```
error[E0107]: missing generics for enum `Option`
   --> src/main.rs:16:17
    |
16  | #[derive(Debug, New)]
    |                 ^^^ expected 1 generic argument
    |
note: enum defined here, with 1 generic parameter: `T`
   --> /home/xxx/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:563:10
    |
563 | pub enum Option<T> {
    |          ^^^^^^ -
    = note: this error originates in the derive macro `New` (in Nightly builds, run with -Z macro-backtrace for more info)
help: add missing generic argument
    |
16  | #[derive(Debug, New<T>)]
```

**Comment by tivrfoa** [comment]
The same happens for `Box<T>` , `Vec<T>`, ...

```
error[E0107]: missing generics for struct `Box`
```

**Comment by tivrfoa** [comment]
Solved by [656b04f](https://github.com/tivrfoa/new-derive/commit/656b04f1e9135dc5ad0e7ed44875681b448389e6)
